comet-java-client/src/main/java/ml/comet/experiment/context/ExperimentContext.java

```diff
@@ -0,0 +1,120 @@
+package ml.comet.experiment.context;
+
+import lombok.Data;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Describes context of the {@link ml.comet.experiment.Experiment}.
+ */
+@Data
+public final class ExperimentContext {
+    private long step;
+    private long epoch;
+    private String context;
+
+    ExperimentContext() {
+    }
+
+    /**
+     * Creates new instance with specified parameters.
+     *
+     * @param step    the current step of the experiment.
+     * @param epoch   the current epoch of the experiment.
+     * @param context the current context of the data log operation.
+     */
+    public ExperimentContext(long step, long epoch, String context) {
+        this.step = step;
+        this.epoch = epoch;
+        this.context = context;
+    }
+
+    /**
+     * Creates new instance with specified parameters.
+     *
+     * @param step  the current step of the experiment.
+     * @param epoch the current epoch of the experiment.
+     */
+    public ExperimentContext(long step, long epoch) {
+        this(step, epoch, StringUtils.EMPTY);
+    }
+
+    /**
+     * Creates new instance with specified parameters.
+     *
+     * @param step the current step of the experiment.
+     */
+    public ExperimentContext(long step) {
```

Comment:
why not use step and epoch as Long (non primitive) so we dont need to default to 0 and we can default to null ?

Reply:
Fixed